test/e2e/deploy/deploy.go

```diff
@@ -1277,18 +1282,33 @@ func (d Deployment) DeleteAllPods(c *Cluster) error {
 	return nil
 }
 
-// LookupCSIAddresses returns controller and node addresses for gRPC dial.
+// LookupCSIAddresses returns controller and node addresses for pod/dial.go (<namespace>.<pod>:<port>).
 // Only works for testing deployments.
-func LookupCSIAddresses(c *Cluster, namespace string) (nodeAddress, controllerAddress string, err error) {
+func LookupCSIAddresses(ctx context.Context, c *Cluster, namespace string) (nodeAddress, controllerAddress string, err error) {
 	// Node #1 is expected to have a PMEM-CSI node driver
 	// instance. If it doesn't, connecting to the PMEM-CSI
-	// node service will fail.
-	nodeAddress = c.NodeServiceAddress(1, SocatPort)
+	// node service will fail. If we only have one node,
+	// then we use that one.
+	node := 1
+	if node >= c.NumNodes() {
+		node = 0
+	}
+	ip := c.NodeIP(node)
+	pod, err := c.GetAppInstance(ctx, labels.Set{"app.kubernetes.io/component": "node-testing"}, ip, namespace)
+	if err != nil {
+		return "", "", fmt.Errorf("find socat pod on node #%d = %s: %v", node, ip, err)
+	}
 
-	// Also use that same node as controller.
-	controllerAddress = nodeAddress
+	for _, port := range pod.Spec.Containers[0].Ports {
+		if port.Name == "csi-socket" {
+			nodeAddress = fmt.Sprintf("%s.%s:%d", namespace, pod.Name, port.ContainerPort)
+			// Also use that same node as controller.
+			controllerAddress = nodeAddress
+			return
 
-	return
+		}
+	}
+	return "", "", fmt.Errorf("container port 'csi-socket' not found in pod %+v", pod)
```

Comment:
This breaks skew testing because older releases didn't have this container port entry.

I'm currently testing a fix.

@avalluri: do you want me to force-push right away or wait until you are done with your review?